Sources/FoundationEssentials/Error/CocoaError+FilePath.swift

```diff
@@ -121,7 +127,68 @@ extension CocoaError {
             userInfo: additionalUserInfo.addingUserInfo(forURL: url)
         )
     }
-    
+
+#if os(Windows)
+    static func errorWithFilePath(_ path: PathOrURL, win32 dwError: DWORD, reading: Bool, variant: String? = nil, userInfo: [String : AnyHashable] = [:]) -> CocoaError {
+        switch path {
+        case let .path(path):
+            return Self.errorWithFilePath(path, win32: dwError, reading: reading, variant: variant, userInfo: userInfo.addingUserInfo(forPath: path))
+        case let .url(url):
+            return Self.errorWithFilePath(url.withUnsafeFileSystemRepresentation { String(cString: $0!) }, win32: dwError, reading: reading, variant: variant, userInfo: userInfo.addingUserInfo(forURL: url))
```

Comment:
Is it worth just using `url.path` here? Or is routing through the file system representation important?

Reply:
I think that the routing through the FSR is important as the error message would give you a path that is not usable otherwise.